### PR TITLE
fix finding 41 - modules can prevent from uninstalling themselves

### PR DESCRIPTION
**Pull request by livingrockrises**
have used try catch

but could also have done it like this

(bool success, ) = hook.call(abi.encodeWithSelector(IModule.onUninstall.selector, data));
        if (!success) {
        // leave this block empty to ignore the revert
        }

### CODE REVIEW

File: contracts/Nexus.sol [hunk 1]
```diff
@@ -222,7 +222,7 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
         require(success, NexusInitializationFailed());
     }
 
-    function setRegistry(IERC7484 newRegistry, address[] calldata attesters, uint8 threshold) external onlyEntryPointOrSelf {
+    function setRegistry(IERC7484 newRegistry, address[] calldata attesters, uint8 threshold) external payable onlyEntryPointOrSelf {
```
[the remark this answers is not in the document]
Comment (livingrockrises) [reply contracts/Nexus.sol:225]: yeah I clubbed. everything else is payable. this will just make consistent and save some gas..

**Comment by livingrockrises** [comment]
I just have one doubt before merging that,
should I even need to return
if(!success) return;

cmichel will confirm this.

**Review by VGabriel45 — APPROVED** [review]

**Review by Aboudjem — APPROVED** [review]
need tests for this

**Comment by livingrockrises** [comment]
> need tests for this

let's make a list of pending tests and still merge today. 
you / I will be working on pending tests in one PR later

**Pull request merged**